### DISCUSSION TITLE
Commands with selectedItem or itemID as CommandParameter

### DISCUSSION THREAD

**Discussion by bsdRuf** [AvaloniaUI/Avalonia#17316] [Q&A] [answered]
Hello, how can I pass the selectedItem or the ID of the binding in a ListBox via ContextMenu?
The [examples ](https://docs.avaloniaui.net/docs/guides/data-binding/how-to-bind-to-a-command-with-reactiveui)have been static parameters. I didnt figure out how to handle ItemsSource related value Binding.

**XAML**
```
<ListBox Name="lb18" ItemsSource="{Binding Lst18}">
      <ListBox.ItemTemplate>
	      <DataTemplate>
			<TextBlock Text="{Binding ItemDesc}"/>
	      </DataTemplate>
      </ListBox.ItemTemplate>
      <ListBox.ContextMenu>
	      <ContextMenu>
		      <MenuItem Command="{Binding DeleteCommand}" CommandParameter="{Binding ItemID}" Header="Delete"/>
	      </ContextMenu>
      </ListBox.ContextMenu>
</ListBox>
```

**Viewmodel**
```
public IReactiveCommand DeleteCommand { get; }
public void DeleteItem(int i)
{
 do sth with the parameter
}
```

Thanks in advance

**Comment by bsdRuf** [comment]
I switched to a diffrent idea to solve it
```
<ListBox ItemsSource="{Binding MyItemList}">
	<ListBox.ItemTemplate>
		<DataTemplate>
			<Grid RowDefinitions="*" ColumnDefinitions="*,Auto" Margin="0,0,0,0">
				<TextBlock Text="{Binding ItemDesc}" VerticalAlignment="Center" HorizontalAlignment="Stretch" Grid.Column="0"/>
				<Button Click="DelItemClicked" DataContext="{Binding ItemId}" Grid.Column="1">
					<Image Source="/Assets/icons/delete.png" Width="12" Height="12" VerticalAlignment="Center" HorizontalAlignment="Right" />
				</Button>
			</Grid>
		</DataTemplate>
	</ListBox.ItemTemplate>
</ListBox>
```

By the DelItemClicked method I could receive the ItemId (out of the sender parameter) to remove it from the list.

**Comment by maxkatz6** [comment, marked as answer]
Command parameter needs to be of `object` type, not integer, see https://learn.microsoft.com/en-us/dotnet/api/system.windows.input.icommand.execute?view=net-8.0#system-windows-input-icommand-execute(system-object).

You can cast it in runtime though, inside of the command, if you always pass an integer there.
